src/scriptInterfaceMagic.h

```diff
@@ -227,21 +227,58 @@ template<typename T> struct convert<std::vector<T> >
             T var;
             convert<T>::param(L, idx, var);
             v.push_back(var);
-        }
-    }
+        }
+    }
+
+    static int returnType(lua_State* L, std::vector<T> vector)
+    {
+        lua_newtable(L);
+        int n = 1;
+        for(auto val : vector)
+        {
+            lua_pushnumber(L, n);
+            if (convert<T>::returnType(L, val))
+            {
+                lua_settable(L, -3);
+                n++;
```

Comment:
lua_seti is a bit easier and less code in this case.

Reply:
cool hint